test/model/message_test.dart

```diff
@@ -524,24 +531,137 @@ void main() {
+          store.reconcileMessages([messageCopy]);
+          checkStoredMessageIdenticalTo(messageCopy);
+          return messageCopy;
+        }
+
+        void checkNoClobber({required Message messageBefore}) {
+          store.reconcileMessages([copyStoredMessage()]);
+          checkStoredMessageIdenticalTo(messageBefore);
+        }
+
+        test('various conditions', () async {
+          final channel = eg.stream();
+          await prepare(stream: channel, isChannelSubscribed: true);
+          final message = eg.streamMessage(id: 1, stream: channel);
+
+          final otherChannel = eg.stream();
+          await store.addStream(otherChannel);
+
+          store.reconcileMessages([message]);
+          checkStoredMessageIdenticalTo(message);
+          // Not clobbering, because the first call didn't mark stale,
+          // because the message was in a subscribed channel.
+          checkNoClobber(messageBefore: message);
+
+          await store.removeSubscription(channel.streamId);
+          // Clobbering because the unsubscribe event marked the message stale.
+          Message messageCopy = checkClobber();
+          // (Check that reconcileMessage itself didn't unmark as stale.)
+          messageCopy = checkClobber();
+
+          await store.addSubscription(eg.subscription(channel));
+          // The channel became subscribed,
+          // but the message's data hasn't been refreshed, so clobber…
+          messageCopy = checkClobber();
+
+          // …Now it's been refreshed, by reconcileMessages, so don't clobber.
+          checkNoClobber(messageBefore: messageCopy);
+
+          check(store.subscriptions[otherChannel.streamId]).isNull();
+          await store.handleEvent(
+            eg.updateMessageEventMoveFrom(origMessages: [message],
+            newStreamId: otherChannel.streamId));
+          // Message was moved to an unsubscribed channel, so clobber.
+          messageCopy = checkClobber(
+            withMessageCopy: copyStoredMessage(displayRecipient: otherChannel.name));
+          // (Check that reconcileMessage itself didn't unmark as stale.)
+          messageCopy = checkClobber();
+
+          // Subscribe, to mark message as not-stale, setting up another check…
+          await store.addSubscription(eg.subscription(otherChannel));
+
+          await store.handleEvent(ChannelDeleteEvent(id: 1, streams: [otherChannel]));
+          // Message was in a channel that became unknown, so clobber.
+          checkClobber();
+        });
+
+        test('in unsubscribed channel on first call', () async {
+          await prepare(isChannelSubscribed: false);
+          final message = eg.streamMessage(id: 1);
+
+          store.reconcileMessages([message]);
+          checkStoredMessageIdenticalTo(message);
+
+          checkClobber();
+          checkClobber();
+        });
+
+        test('new-message event when in unsubscribed channel', () async {
```

Comment:
Another case to test: message event in subscribed channel

Reply:
This test is now quite easy to write! 😄

```dart
        test('new-message event when in a subscribed channel', () async {
          await prepare(isChannelSubscribed: true);
          final message = eg.streamMessage();

          await store.handleEvent(eg.messageEvent(message));

          checkNoClobber();
          checkNoClobber();
        });
```